src/utils/allocators.zig

```diff
@@ -320,3 +320,59 @@ test "disk allocator on arraylists" {
     };
     try std.testing.expect(did_error);
 }
+
+pub const failing = struct {
```

Comment:
dead code?

Reply:
It's dead, but potentially very useful code - it's (imo) a more useful version of `std.testing.failing_allocator`, which would even be reasonable to use outside of tests for the purposes of asserting that an allocating path is not reached.
I can remove it if you want, and just save it until such a time where we need it.

Reply:
in that case - maybe add a comment explaining context around how its useful and how it could be used

Reply:
Well, there is a doc comment, this is just the namespace in which it lives. This seemed better than having to prefix everything related to it with `FailingAllocator`:
* `utils.allocators.FailingAllocatorConfig` vs `utils.allocators.failing.Config`
* `utils.allocators.FailingAllocatorMode` vs `utils.allocators.failing.Mode`
* `utils.allocators.failingAllocator` vs `utils.allocators.failing.allocator`

Reply:
it makes sense but is different from everywhere else in the codebase which leads to confusion

the config/mode isnt that big of deal since they wont be imported much -- you can just do `FailingAllocator.allocator(.{ .alloc = .noop_or_fail, ... })`

would rather have it follow how everywhere else does it (including how zig's GPA does it)

Reply:
The stdlib does it that way for gpa because gpa isn't a stateless allocator - a better comparison would be to `c_allocator` or `page_allocator`: there is no `cAllocator(){}.allocator()`, you simply access the global constant for the `std.mem.Allocator` interface. The same occurs here, it is simply a parameterized global.

As for the namespace grouping, that is a very common convention throughout both the ecosystem and in the stdlib itself, and is a very useful way to organize code. It also facilitates being able to pull in an entire group of related declarations without having to reference each individual one.

I don't see any scenario where this could realistically lead to any real confusion: it's the same as if it were in another file and we exported the namespace here.

Reply:
43f2956f225d009a3583fd69c6e8502ae1912df6